### ISSUE TITLE
Can't connect google calendar 

### ISSUE THREAD

**Issue by dzania** [calcom/cal.com#9816]
### Issue Summary

When i try to add google calendar during user setup I got ERR_INVALID_URL in logs
``` 
@calcom/web:start:     at apiResolver (/calcom/node_modules/next/dist/server/api-utils/node.js:461:63) {
2023-06-27T07:19:00.041826908Z @calcom/web:start:   input: '/api/collect-events',
2023-06-27T07:19:00.041869007Z @calcom/web:start:   code: 'ERR_INVALID_URL'
@calcom/web:start: }
@calcom/web:start: 2023-06-27 07:19:00.063 UTC [ERROR] (next-collect/server) - Invalid URL TypeError [ERR_INVALID_URL]: Invalid URL
)
@calcom/web:start:     at ContextAPI.with (/calcom/node_modules/@opentelemetry/api/build/src/api/context.js:60:46)
@calcom/web:start:     at NoopTracer.startActiveSpan (/calcom/node_modules/@opentelemetry/api/build/src/trace/NoopTracer.js:65:31)
@calcom/web:start:     at ProxyTracer.startActiveSpan (/calcom/node_modules/@opentelemetry/api/build/src/trace/ProxyTracer.js:36:24)
@calcom/web:start:     at /calcom/node_modules/next/dist/server/lib/trace/tracer.js:106:107
@calcom/web:start:     at NoopContextManager.with (/calcom/node_modules/@opentelemetry/api/build/src/context/NoopContextManager.js:25:19)
2023-06-27T07:19:00.066654652Z @calcom/web:start:     at ContextAPI.with (/calcom/node_modules/@opentelemetry/api/build/src/api/context.js:60:46)
@calcom/web:start:     at NextTracerImpl.trace (/calcom/node_modules/next/dist/server/lib/trace/tracer.js:106:32)
@calcom/web:start:     at apiResolver (/calcom/node_modules/next/dist/server/api-utils/node.js:461:63) {
2023-06-27T07:19:00.066741607Z @calcom/web:start:   input: '/api/collect-events',
@calcom/web:start:   code: 'ERR_INVALID_URL'
````

### Steps to Reproduce

1. Deploy a self hosted instance and start user creation
2. Try to connect google calendar during user creation

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

### Technical details

Using latest docker image version which is 3.0.7

**Comment by PeerRich** [comment]
just to confirm, you have the correct google credentials?

**Comment by siddjellali** [comment]
Hi ! 
Seems to be the same issue here on the lastest release.

```
@calcom/web:start: 2023-09-26 17:44:07.964 UTC [ERROR] (next-collect/server) - Invalid URL TypeError [ERR_INVALID_URL]: Invalid URL
@calcom/web:start:     at new NodeError (node:internal/errors:405:5)
@calcom/web:start:     at new URL (node:internal/url:611:13)
@calcom/web:start:     at Object.parsePublicUrl (/calcom/node_modules/next-collect/dist/next-shim.js:74:27)
@calcom/web:start:     at Object.nextApiHandler (/calcom/node_modules/next-collect/dist/server.js:58:40)
@calcom/web:start:     at /calcom/node_modules/next-collect/dist/server.js:180:25
@calcom/web:start:     at /calcom/node_modules/next/dist/server/api-utils/node.js:463:16
@calcom/web:start:     at /calcom/node_modules/next/dist/server/lib/trace/tracer.js:117:36
@calcom/web:start:     at NoopContextManager.with (/calcom/node_modules/@opentelemetry/api/build/src/context/NoopContextManager.js:25:19)
@calcom/web:start:     at ContextAPI.with (/calcom/node_modules/@opentelemetry/api/build/src/api/context.js:60:46)
@calcom/web:start:     at NoopTracer.startActiveSpan (/calcom/node_modules/@opentelemetry/api/build/src/trace/NoopTracer.js:65:31)
@calcom/web:start:     at ProxyTracer.startActiveSpan (/calcom/node_modules/@opentelemetry/api/build/src/trace/ProxyTracer.js:36:24)
@calcom/web:start:     at /calcom/node_modules/next/dist/server/lib/trace/tracer.js:106:107
@calcom/web:start:     at NoopContextManager.with (/calcom/node_modules/@opentelemetry/api/build/src/context/NoopContextManager.js:25:19)
@calcom/web:start:     at ContextAPI.with (/calcom/node_modules/@opentelemetry/api/build/src/api/context.js:60:46)
@calcom/web:start:     at NextTracerImpl.trace (/calcom/node_modules/next/dist/server/lib/trace/tracer.js:106:32)
@calcom/web:start:     at apiResolver (/calcom/node_modules/next/dist/server/api-utils/node.js:461:63) {
@calcom/web:start:   input: '/api/collect-events',
@calcom/web:start:   code: 'ERR_INVALID_URL'
@calcom/web:start: }
```

I can connect my calendar but i have some disconnection.

![image](https://github.com/calcom/cal.com/assets/3965823/21d24618-0ad5-42df-81be-d59a06b1768f)

**Comment by siddjellali** [comment]
fixed by v3.3.3

**Comment by dzania** [comment]
fixed by v3.3.3
